editor/plugins/node_3d_editor_plugin.cpp

```diff
@@ -6031,6 +6038,42 @@ void Node3DEditor::_update_gizmos_menu_theme() {
 	}
 }
 
+Vector3 Node3DEditor::_get_perspective_position() {
```

Comment:
There's something going on with this function... If I read the code correctly, it only returns the intersecting camera position of the last enabled grid axis. So if all 3 axis are enabled, 2 of them will be ignored. 

It looks like we have to handle the 3 axis separately when the camera is in orthogonal mode, so store the 3 projected positions separately and update the grids whenever one of them moves too far. We could even update just the axis that need it, but I wouldn't bother with that.

Also, the `EditorSettings::get_singleton()->get()` calls are expensive, so we could cache them or at least use `SNAME()` on the property names to avoid re-creating the string names on every frame, and only do the separate axis checks whenever the camera is in orthogonal mode.

Reply:
one sec I'll confirm if we can't use the position directly I may have made an error

Reply:
code sis sus and I thought it too

Reply:
get_position works in ortho mode too fine 👍 thanks for pointing that out I'll remove it and just use position again, it must have been just needing the redraw